File: AspNetMvcUnleashed/0672329980_source/Chapter20/CS/UnleashedBlog/UnleashedBlog/Controllers/ArchiveController.cs
using System.Web.Mvc;
using UnleashedBlog.Models;
using UnleashedBlog.Models.EntityFramework;
using UnleashedBlog.Attributes;

namespace UnleashedBlog.Controllers
{
    public class ArchiveController : Controller
    {
        private BlogServiceBase _blogService;

        public ArchiveController()
        {
            _blogService = new BlogService(this.ModelState);
        }

        public ArchiveController(BlogRepositoryBase blogRepository)
        {
            _blogService = new BlogService(this.ModelState, blogRepository);
        }

        public ActionResult Index(int? page, int? year, int? month, int? day, string name)
        {
            return View(_blogService.ListBlogEntries(page, year, month, day, name));
        }

        [AcceptAjax]
        [ActionName("Index")]
        public ActionResult Index_Ajax(int? page, int? year, int? month, int? day, string name)
        {
            return PartialView("BlogEntries", _blogService.ListBlogEntries(page, year, month, day, name));
        }


    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter20/VB/UnleashedBlog/UnleashedBlog/Controllers/ArchiveController.vb
Imports UnleashedBlog.Attributes

Public Class ArchiveController
    Inherits Controller

    Private _blogService As BlogServiceBase

    Public Sub New()
        _blogService = New BlogService(Me.ModelState)
    End Sub

    Public Sub New(ByVal blogRepository As BlogRepositoryBase)
        _blogService = New BlogService(Me.ModelState, blogRepository)
    End Sub

    Public Function Index(ByVal page As Integer?, ByVal year As Integer?, ByVal month As Integer?, ByVal day As Integer?, ByVal name As String) As ActionResult
        Return View(_blogService.ListBlogEntries(page, year, month, day, name))
    End Function

    <AcceptAjax(), ActionName("Index")> _
    Public Function Index_Ajax(ByVal page As Integer?, ByVal year As Integer?, ByVal month As Integer?, ByVal day As Integer?, ByVal name As String) As ActionResult
        Return PartialView("BlogEntries", _blogService.ListBlogEntries(page, year, month, day, name))
    End Function

End Class
